contracts/L1GlobalForkRequester.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.20;
+
+/* 
+   Contract to proxy a fork request from the bridge to its ForkingManager
+   Any L2 contract can call us over the bridge to get us to request a fork.
+   We record the dispute they were forking over
+*/
+
+// TODO: An alternative would be to look up the ForkingManager on L2ChainInfo and call it directly.
+// TODO: We could use this to manage the dispute data, in which case we could gate initiateFork() to only be callable by us.
+// TODO: The whitepaper implies a whitelist about who is allowed to fork us. Currently anybody can as long as they pay.
+
+import {IForkingManager} from "./interfaces/IForkingManager.sol";
+import {IPolygonZkEVMBridge} from "@RealityETH/zkevm-contracts/contracts/interfaces/IPolygonZkEVMBridge.sol";
+import {IPolygonZkEVM} from "@RealityETH/zkevm-contracts/contracts/interfaces/IPolygonZkEVM.sol";
+
+// NB We'd normally use the interface IForkableBridge here but it causes an error:
+//  Error (5005): Linearization of inheritance graph impossible
+import {ForkableBridge} from "./ForkableBridge.sol";
```

Comment:
It works perfectly fine, if you add:
import {IForkableStructure} from "./interfaces/IForkableStructure.sol";
and then 
      IForkableStructure bridge = IForkableStructure(msg.sender);
      IForkingManager forkingManager = IForkingManager(bridge.forkmanager());

compiles smoothly, I checked it.

Reply:
Not sure that's any better than just importing ForkableBridge. Instead of being able to declare it as the type it is, you have to keep changing its type whenever you call something against it...